containers.go

```diff
@@ -265,6 +263,8 @@ func newContainerStore(dir string) (ContainerStore, error) {
 		bylayer:    make(map[string]*Container),
 		byname:     make(map[string]*Container),
 	}
+	cstore.Lock()
+	defer cstore.Unlock()
 	if err := cstore.Load(); err != nil {
```

Comment:
It's somehow surprising that callers of a public API are responsible for locking.

Reply:
The `{container,image,layer}Store` objects implement public interfaces, but AFAICS there is no public way to access them (one would have to take a `storage.Store` and manually cast it to an interface that contains accessors to the three stores).

Compare #1345 .

I think the basic idea is that various `storage.Store` operations need to coordinate _multiple_ operations on a single `layerStore`, so they need a consistent view, and a lock held across the `layerStore` calls.

See #1346 which tries to make the locking requirements much more explicit to make sure there are no mistakes there.